### PR TITLE
lib: fix divmod signs

### PR DESCRIPTION
**Pull request by indutny**
Return negative `mod` to make sure that:

    a = Q * b + R

Fix: #47

### CODE REVIEW

**Comment by Yaffle** [comment]
this branch passes my tests

File: test/red-test.js [hunk 1]
```diff
@@ -72,10 +72,23 @@ describe('BN.js/Reduction context', function() {
         var p = new BN(257);
         var m = fn(p);
         var a = new BN(3).toRed(m);
-        var b = a.redInvm(p);
+        var b = a.redInvm();
         assert.equal(a.redMul(b).fromRed().toString(16), '1');
       });
 
+      it('should invm numbers (regression', function() {
```
Comment (dcousens) [inline test/red-test.js]: Missing )

**Comment by indutny** [comment]
@dcousens fixed

**Comment by indutny** [comment]
@dcousens how about this? Should fix all mentioned issues (by throwing :D )

**Comment by dcousens** [comment]
@indutny hahaha.
That works,  just are you sure you want to expose the unsigned methods?  That means supporting them later.

**Comment by indutny** [comment]
@dcousens I'm going to use them in elliptic. They appear to be very useful in the reduction-only case. Supporting them does not seem to be a big problem, considering that signed versions are most likely going to fall back to the unsigned ones in some (if not all) cases.

**Comment by dcousens** [comment]
Solid,  LGTM :+1:

File: lib/bn.js [hunk 2]
```diff
@@ -609,9 +626,16 @@ BN.prototype.and = function and(num) {
...
+    return this.clone().iuand(num);
+  else
+    return num.clone().iuand(this);
+};
+
 
 // Xor `num` with `this` in-place
-BN.prototype.ixor = function ixor(num) {
+BN.prototype.iuxor = function iuxor(num) {
   this.sign = this.sign || num.sign;
```
Comment (dcousens) [inline lib/bn.js]: Given that this is `unsigned`,  does this logic still make sense?

Comment (indutny) [inline lib/bn.js]: It doesn't, thanks.

Comment (indutny) [inline lib/bn.js]: Fixed.

**Comment by indutny** [comment]
Landed and released! Thank you!

**Comment by ashnur** [comment]
This bit me.

**Comment by indutny** [comment]
@ashnur how come? It is fixed now.

**Comment by ashnur** [comment]
This was written for 2.0.5 https://github.com/rat-nest/big-rat/blob/master/lib/num-to-bn.js#L13 and when I updated to latest it broke. I haven't written the original, so I am not sure if it was used correctly. I rewrote the code to use `.ushln` instead which passes the tests, but it will take me some time to be sure if everything is correct.

**Comment by indutny** [comment]
Aaah, I see now. I'm terribly sorry for this, but it has to be done this way.

**Pull request merged**